.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,181 @@
+name: integration-tests
+on:
+  # Use a manual approval process before PR's are given access to
+  # the secrets which are required to run the integration tests.
+  # The PR code should be manually approved to see if it can be trusted.
+  # When in doubt, do not approve the test run.
+  # Reference: https://dev.to/petrsvihlik/using-environment-protection-rules-to-secure-secrets-when-building-external-forks-with-pullrequesttarget-hci
+  pull_request_target:
+    branches: [ main, add-robot-libraries ]
+  push:
+    branches: [ main, add-robot-libraries ]
```

Comment:
So, to run the integration tests one has to push it to one of these branches right? Then how they will run on every PR?

Reply:
Good catch, the `add-robot-libraries` is only activated for my PR testing (to ensure the workflow actually worked). I wanted to leave it in place until I have demoed the new testing setup to the team, then delete it before merging. I will use this conversation as a TODO.

[ ] Remove `add-robot-libraries` from the `integration-test` workflow